src/thunderbots/software/ai/intent/avoid_area.h

```diff
@@ -2,30 +2,59 @@
 
 #include <ostream>
 
+#include "ai/world/world.h"
+
 /**
  * This enum describes the areas that an intent can choose to avoid. They serve
  * as an indicator to the navigator when it interprets the intents
  */
 enum class AvoidArea
 {
-    // The enemy robots and their avoid obstacles
-    ENEMY_ROBOTS,
+    // Used for iterating over avoid areas
+    FIRST_AVOID_AREA = 1 << 0,
     // The box around the friendly goal
-    FRIENDLY_DEFENSE_AREA,
+    FRIENDLY_DEFENSE_AREA = 1 << 0,
     // The box around the enemy goal
-    ENEMY_DEFENSE_AREA,
+    ENEMY_DEFENSE_AREA = 1 << 1,
     // The box around the enemy goal, inflated by 0.2 meters for certain situations
-    INFLATED_ENEMY_DEFENSE_AREA,
+    INFLATED_ENEMY_DEFENSE_AREA = 1 << 2,
     // The center circle
-    CENTER_CIRCLE,
+    CENTER_CIRCLE = 1 << 3,
     // A half meter radius around the ball
-    HALF_METER_AROUND_BALL,
+    HALF_METER_AROUND_BALL = 1 << 4,
     // The ball itself
-    BALL,
+    BALL = 1 << 5,
     // The enemy half of the field
-    ENEMY_HALF,
+    ENEMY_HALF = 1 << 6,
     // The friendly half of the field
-    FRIENDLY_HALF
+    FRIENDLY_HALF = 1 << 7,
+    // Used for iterating over avoid areas
+    LAST_AVOID_AREA_BEFORE_ROBOTS = 1 << 7,
+    // The enemy robots and their avoid obstacles
+    ENEMY_ROBOTS = 1 << 8,
+    // Used for iterating over avoid areas
+    LAST_AVOID_AREA = 1 << 8,
 };
 
+typedef uint32_t avoid_area_mask_t;
```

Comment:
Maybe `std::bitset<N>` might be better to use as it already implements set/reset and flip so we don't need to roll out our own, it might make everything more readable as you would then say `avoid_areas.set(AvoidArea::FRIENDLY_HALF)`, rather than `MERGE_AVOID_AREAS(avoid_area, AvoidArea::FRIENDLY_HALF)`

Reply:
ok